src/python/grpcio/grpc/_cython/_cygrpc/aio/callback_common.pyx.pxi

```diff
@@ -35,6 +35,8 @@ cdef class CallbackWrapper:
     def __cinit__(self, object future, CallbackFailureHandler failure_handler):
         self.context.functor.functor_run = self.functor_run
         self.context.waiter = <cpython.PyObject*>future
+        # TODO(lidiz) switch to future.get_loop() which is available 3.7+.
+        self.context.loop = <cpython.PyObject*>future._loop
```

Comment:
Technically we could and maybe we should make the plumbing of the loop at the up call level, so instead of inferring the loop by accessing the none public attribute create the context by passing as a parameter the loop. 

It's a pity that `future.get_loop()` is only available in Python versions 3.7 and beyond.

But TBH we might take the risk since we are gonna support only Python 3.6 and its hard to believe that this internal attribute will be changed in patch release of the 3.6. But then we would need to do something like:

```python
if python_version > (3, 6):
    loop = future.get_loop()
else:
    loop = future._loop
```

Reply:
Changed to plumb through the `loop` object. Passing a reference is cheaper than compare a Python object.